README_vulkan.md

```diff
@@ -0,0 +1,569 @@
+# Vulkan Support for Halide
+
+Halide supports the Khronos Vulkan framework as a compute API backend for GPU-like 
+devices, and compiles directly to a binary SPIR-V representation as part of its 
+code generation before submitting it to the Vulkan API. Both JIT and AOT usage 
+are supported via the `vulkan` target flag (eg `HL_JIT_TARGET=host-vulkan`).
+
+Vulkan support is actively under development, and considered *EXPERIMENTAL*
+at this stage.  Basic tests are passing, but there's still work to do to
+until we have adequate feature parity for production use.  
+
+See [below](#current-status) for details on specific test cases.
+
+# Compiling Halide w/Vulkan Support
+
+You'll need to configure Halide and enable the cmake option TARGET_VULKAN.
+
+For example, on Linux & OSX:
+
+```
+% cmake -G Ninja -DTARGET_VULKAN=ON -DCMAKE_BUILD_TYPE=Release -DLLVM_DIR=$LLVM_ROOT/lib/cmake/llvm -S . -B build
+% cmake --build build
+```
+
+# Vulkan Runtime Environment:
+
+Halide has no direct dependency on Vulkan for code-generation, but the runtime
+requires a working Vulkan environment to run Halide generated code. Any valid 
+Vulkan v1.0+ device driver should work. 
+
+Specifically, you'll need:
+
+-   A vendor specific Vulkan device driver
+-   The generic Vulkan loader library
+
+For AMD & NVIDIA & Intel devices, download and install the latest graphics driver 
+for your platform. Vulkan support should be included.
+
+## Windows 
+
+AMD:
+https://www.amd.com/en/technologies/vulkan
+
+NVIDIA:
+https://developer.nvidia.com/vulkan-driver
+
+INTEL:
+https://www.intel.com/content/www/us/en/download-center/home.html
+
+
+## Linux 
+
+On Ubuntu Linux, proprietary drivers can be installed via 'apt' using 
+PPA's for each vendor.
+
+For AMD:
+```
+$ sudo add-apt-repository ppa:oibaf/graphics-drivers
+$ sudo apt update
+$ sudo apt upgrade
+$ sudo apt install libvulkan1 mesa-vulkan-drivers vulkan-utils
+```
+
+For NVIDIA:
+```
+$ sudo add-apt-repository ppa:graphics-drivers/ppa
+$ sudo apt update
+$ sudo apt upgrade
+# - replace ### with latest driver release (eg 515)
+$ sudo apt install nvidia-driver-### nvidia-settings vulkan vulkan-utils
+```
+
```

Comment:
For Ubuntu 22.04, these steps didn't quite work for me.  The last step I needed to do was:

```
sudo apt install nvidia-driver-515 nvidia-settings libvulkan vulkan-tools
```

After that, I can run tutorials or a test with `HL_JIT_TARGET=host-vulkan` successfully.  However, building with `-DHalide_TARGET=host-vulkan` doesn't work due to link errors:
```
[3094/3631] Linking CXX executable test/generator/generator_aot_can_use_target
FAILED: test/generator/generator_aot_can_use_target
: && /usr/bin/c++   test/generator/CMakeFiles/generator_aot_can_use_target.dir/can_use_target_aottest.cpp.o -o test/generator/generator_aot_can_use_target  test/generator/libcan_use_target.a  test/generator/libcan_use_target.runtime.a  -ldl && :
/usr/bin/ld: test/generator/libcan_use_target.runtime.a(can_use_target.runtime.o): in function `halide_vulkan_acquire_context':
posix_allocator.cpp:(.text.halide_vulkan_acquire_context+0x1d8): undefined reference to `vkGetInstanceProcAddr'
/usr/bin/ld: posix_allocator.cpp:(.text.halide_vulkan_acquire_context+0x520): undefined reference to `vkCreateInstance'
/usr/bin/ld: posix_allocator.cpp:(.text.halide_vulkan_acquire_context+0x103a): undefined reference to `vkGetInstanceProcAddr'
/usr/bin/ld: posix_allocator.cpp:(.text.halide_vulkan_acquire_context+0x104e): undefined reference to `vkGetInstanceProcAddr'
/usr/bin/ld: test/generator/libcan_use_target.runtime.a(can_use_target.runtime.o): in function `Halide::Runtime::Internal::Vulkan::vk_load_vulkan_functions(VkInstance_T*)':
posix_allocator.cpp:(.text._ZN6Halide7Runtime8Internal6Vulkan24vk_load_vulkan_functionsEP12VkInstance_T+0x11): undefined reference to `vkGetInstanceProcAddr'
/usr/bin/ld: posix_allocator.cpp:(.text._ZN6Halide7Runtime8Internal6Vulkan24vk_load_vulkan_functionsEP12VkInstance_T+0x2a): undefined reference to `vkGetInstanceProcAddr'
/usr/bin/ld: posix_allocator.cpp:(.text._ZN6Halide7Runtime8Internal6Vulkan24vk_load_vulkan_functionsEP12VkInstance_T+0x43): undefined reference to `vkGetInstanceProcAddr'
/usr/bin/ld: test/generator/libcan_use_target.runtime.a(can_use_target.runtime.o):posix_allocator.cpp:(.text._ZN6Halide7Runtime8Internal6Vulkan24vk_load_vulkan_functionsEP12VkInstance_T+0x5c): more undefined references to `vkGetInstanceProcAddr' follow
collect2: error: ld returned 1 exit status
```

Reply:
Ahhh ... thanks!  I'll add vulkan-tools to the docs!

Hrmm ... I'll look into the linker issues ... looks like the vulkan loader isn't getting added to the linkage for AOT generators. =/